compile.sh

```diff
@@ -10,9 +10,9 @@ MDIR="./modules"
 TDIR="./test"
 SDIR="./scripts"
 
-gcc -o main.o -I ${IDIR} ${MDIR}/geometry.c ${MDIR}/hamiltonian.c ${MDIR}/powermethod.c ${MDIR}/observables.c ${MDIR}/linearalgebra.c ${MDIR}/conjugategradient.c main.c -lm
-gcc -o ${TDIR}/analytic.o -I ${IDIR} ${MDIR}/geometry.c ${MDIR}/hamiltonian.c ${MDIR}/linearalgebra.c ${MDIR}/assert.c ${TDIR}/analytic.c -lm
-gcc -o ${TDIR}/geometrytest.o -I ${IDIR} ${MDIR}/geometry.c ${TDIR}/geometrytest.c -lm
-gcc -o ${TDIR}/hermitian.o -I ${IDIR} ${MDIR}/geometry.c ${MDIR}/hamiltonian.c ${MDIR}/wavefunction.c ${MDIR}/linearalgebra.c ${MDIR}/assert.c ${TDIR}/hermitian.c -lm
-gcc -o ${TDIR}/power_algo.o -I ${IDIR} ${MDIR}/geometry.c ${MDIR}/powermethod.c ${MDIR}/assert.c  ${MDIR}/linearalgebra.c  ${TDIR}/power_algo.c -lm
-gcc -o ${TDIR}/test_conjugate.o ${TDIR}/test_conjugate.c -lm
+gcc -o main.o -Wall -I ${IDIR} ${MDIR}/geometry.c ${MDIR}/hamiltonian.c ${MDIR}/powermethod.c ${MDIR}/observables.c ${MDIR}/linearalgebra.c ${MDIR}/conjugategradient.c main.c -lm
+gcc -o ${TDIR}/analytic.o -Wall -I ${IDIR} ${MDIR}/geometry.c ${MDIR}/hamiltonian.c ${MDIR}/linearalgebra.c ${MDIR}/assert.c ${TDIR}/analytic.c -lm
+gcc -o ${TDIR}/geometrytest.o -Wall -I ${IDIR} ${MDIR}/geometry.c ${TDIR}/geometrytest.c -lm
+gcc -o ${TDIR}/hermitian.o -Wall -I ${IDIR} ${MDIR}/geometry.c ${MDIR}/hamiltonian.c ${MDIR}/wavefunction.c ${MDIR}/linearalgebra.c ${MDIR}/assert.c ${TDIR}/hermitian.c -lm
+gcc -o ${TDIR}/power_algo.o -Wall -I ${IDIR} ${MDIR}/geometry.c ${MDIR}/powermethod.c ${MDIR}/assert.c  ${MDIR}/linearalgebra.c  ${TDIR}/power_algo.c -lm
+gcc -o ${TDIR}/test_conjugate.o -Wall ${TDIR}/test_conjugate.c -lm
```

Comment:
I did a few more changes on this, you can leave that change out of the merge request too.